README.md

```diff
@@ -181,8 +181,11 @@ try {
 }
 ```
 
-This mechanism could even be codified into `import()` by making it reject if
-the target module has not run to end of source text in Evaluation yet. At the current time [a search](https://github.com/search?utf8=%E2%9C%93&q=%22export+async+function%22&type=Code) for "export async function" on github produces over 5000 unique code examples of exporting an async function.
+While it is tempting to codify this behavior by specifying that `import()` should reject if the imported module has not finished evaluating, that behavior would make `import()` a poor abstraction for importing asynchronous modules, since it would deprive well-behaved modules of the chance to finish async work, just because they were imported using `import()`.
+
+Instead of `import()` rejecting if the imported module is suspended on an `await` expression, it might be more useful if there was a way for dynamic `import()` to obtain a reference to the incomplete module namespace object, similar to how CommonJS `require` returns an incomplete `module.exports` object in the event of cycles. As long as the incomplete namespace object provides enough information, or the namespace object isn't used until later, this strategy would allow the application to keep making progress, rather than deadlocking.
```

Comment:
this would be problematic for a lot of cases, while CJS allowed partial namespaces, it often did not impose TDZs like much of the new syntax such as `let`/`const`/`class` making the namespace be less useful and more prone to throwing errors on attempted use prior to completion of source text evaluation.

Reply:
Of course ESM will have an easier time than CJS of preventing misuse of uninitialized bindings. However, I believe there needs to be a way for cyclic ESM imports (both static and dynamic) to access exports that have already been initialized, even if not all exports are fully initialized. This is the essence of any strategy for mitigating import cycles, in both CJS and ESM.

Reply:
Can you explain the exact workflow you are using to mitigate import cycles with incomplete namespaces, that might clear this up. It seems you might not be avoiding cycles, but have some workflow to avoid deadlock.